app/javascript/components/cloud-volume-form/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './cloud-volume-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const CloudVolumeForm = ({ recordId }) => {
+  const [{ fields, initialValues, isLoading }, setState] = useState({ fields: [], isLoading: !!recordId });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+  const promise = useMemo(() =>
+    API.get('/api/providers?expand=resources&attributes=id,name,supports_block_storage&filter[]=supports_block_storage=true'), []);
+
+  const loadSchema = (value, appendState = {}) => {
+    console.log(value);
+    setState(state => ({
+      ...state,
+      ...appendState,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/providers/${recordId}`).then((initialValues) => {
+        promise.then(loadSchema(initialValues.type, { initialValues, isLoading: false }));
```

Comment:
You will have to chain two requests here, one loading the schema calling an OPTIONS request and one that loads the data as you do. The results of the two calls should make the argument of the `loadSchema` function.

I know this is hard to do without the API, so I'll push on that side or help you set up a custom API branch. Until then just consider this as a note...